### PR TITLE
Restrict dataset download size, update Gems, Datacite transition

### PR DESCRIPTION
**Pull request by whunter**
restrict dataset downloads to 3GB

point host to secrets in ezid config

refer to doi not ezid in secrets. change default host

ezid to doi in ability model

pin jquery back because of autocomplete errors

pin pg

update base path to datacite for doi's

### CODE REVIEW

File: app/controllers/collections_controller.rb [hunk 1]
```diff
@@ -13,6 +13,24 @@ def form_class
     MyCollectionEditForm
   end
 
+  def show
+    super
+    max_dataset_for_export = 3 * 1024 * 1024 * 1024
+    @dataset_size = 0
+    @can_export = true
+    if @collection.members.respond_to?('each')
+      @collection.members.each do |gf|
```
Comment (yinlinchen) [inline app/controllers/collections_controller.rb]: Please use a meaningful variable name, instead of "gf"

**Comment by whunter** [comment]
@yinlinchen changed to "generic_file"

File: app/views/collections/_show_actions.html.erb [hunk 2]
```diff
@@ -8,8 +8,12 @@
...
           <span class="glyphicon glyphicon-file"></span> Files
         <% end %>
-        <%= link_to collections_download_path(@collection), class: "btn btn-default", title: "Export this Dataset" do %>
-          <span class="glyphicon glyphicon-download"></span> Export dataset
+        <% if @can_export %>
+          <%= link_to collections_download_path(@collection), class: "btn btn-default", title: "Export this Dataset" do %>
+            <span class="glyphicon glyphicon-download"></span> Export dataset
+          <% end %>
+        <% else %>
+          <div style="clear:both">The size of this dataset is too large to export or is unknown.</div>
```
Comment (yinlinchen) [inline app/views/collections/_show_actions.html.erb]: Please add  "The maximum export dataset file size is limited to 3GB" or something like that.

File: app/controllers/collections_controller.rb [hunk 3]
```diff
@@ -13,6 +13,24 @@ def form_class
...
+      @collection.members.each do |generic_file|
+        if generic_file.file_size.blank?
+          @can_export = false
+          break
+        end
+        file_size = generic_file.file_size.first.to_i
+        @dataset_size += file_size
+      end
+    end
+    @can_export = (@can_export && @dataset_size < max_dataset_for_export)
```
Comment (yinlinchen) [inline app/controllers/collections_controller.rb]: Is 3GB file downloadable? Should it be 
```
@can_export && @dataset_size <= max_dataset_for_export
```

File: app/views/collections/_show_actions.html.erb (hunk 2 above)
Comment (whunter) [reply app/views/collections/_show_actions.html.erb]: @yinlinchen Added this.

File: app/controllers/collections_controller.rb (hunk 3 above)
Comment (whunter) [reply app/controllers/collections_controller.rb]: 3gb has worked in testing. Files that are much larger cause problems.

**Pull request merged**